File: process.js
const Jimp = require("jimp");
const fs = require("fs");
const path = require("path");

const basePath =
  "C:\\\\Users\\\\Bushra\\\\OneDrive\\\\Desktop\\\\techfest\\\\tech-shastra\\\\apps\\\\web\\\\public\\\\planets";
const images = {
  mercury: "[3D] Sculpted Mercury.jpg",
  saturn: "3D Models _ Over A Million Models For Download _ TurboSquid.jpg",
  venus: "Venus.jpg",
};

async function processImages() {
  for (const [planet, filename] of Object.entries(images)) {
    const filepath = path.join(basePath, filename);
    if (!fs.existsSync(filepath)) continue;
    try {
      const img = await Jimp.read(filepath);
      img.scan(0, 0, img.bitmap.width, img.bitmap.height, function (x, y, idx) {
        const r = this.bitmap.data[idx + 0];
        const g = this.bitmap.data[idx + 1];
        const b = this.bitmap.data[idx + 2];
        if (r > 230 && g > 230 && b > 230) {
          this.bitmap.data[idx + 3] = 0; // Alpha to 0
        }
      });
      img.autocrop();
      const outPath = path.join(basePath, planet + ".png");
      await img.writeAsync(outPath);
      console.log("Saved " + outPath);
    } catch (e) {
      console.error("Error processing " + filename, e);
    }
  }

  // Update timeline.jsx
  const jsxPath =
    "C:\\\\Users\\\\Bushra\\\\OneDrive\\\\Desktop\\\\techfest\\\\tech-shastra\\\\apps\\\\web\\\\src\\\\Pages\\\\timeline.jsx";
  let code = fs.readFileSync(jsxPath, "utf-8");
  code = code.replace(
    /mercury:\s*"data:[^"]+",?/,
    'mercury: "/planets/mercury.png",'
  );
  code = code.replace(
    /venus:\s*"data:[^"]+",?/,
    'venus: "/planets/venus.png",'
  );
  code = code.replace(
    /saturn:\s*"data[^"]+",?/,
    'saturn: "/planets/saturn.png",'
  );
  // Ensure saturn exists in IMG object if it was missing:
  if (!code.includes('saturn: "/planets/saturn.png"')) {
    code = code.replace(
      /jupiter:\s*"data:[^"]+",?/,
      'jupiter: "data...",\n    saturn: "/planets/saturn.png",'
    );
  }
  fs.writeFileSync(jsxPath, code, "utf-8");
  console.log("Updated timeline.jsx");
}
processImages();

File: process.py
import os
import re
from PIL import Image

images = {
    'mercury': '[3D] Sculpted Mercury.jpg',
    'saturn': '3D Models _ Over A Million Models For Download _ TurboSquid.jpg',
    'venus': 'Venus.jpg'
}

base_path = r'C:\Users\Bushra\OneDrive\Desktop\techfest\tech-shastra\apps\web\public\planets'

for planet, filename in images.items():
    filepath = os.path.join(base_path, filename)
    if not os.path.exists(filepath):
        print(f"File not found: {filename}")
        continue
    try:
        img = Image.open(filepath).convert("RGBA")
        datas = img.getdata()
        newData = []
        
        for item in datas:
            if item[0] > 230 and item[1] > 230 and item[2] > 230:
                newData.append((255, 255, 255, 0))
            else:
                newData.append(item)
                
        img.putdata(newData)
        bbox = img.getbbox()
        if bbox:
            img = img.crop(bbox)
            
        out_path = os.path.join(base_path, f'{planet}.png')
        img.save(out_path, "PNG")
        print(f"Saved {out_path}")
    except Exception as e:
        print(f"Error processing {filename}: {e}")

jsx_path = r'C:\Users\Bushra\OneDrive\Desktop\techfest\tech-shastra\apps\web\src\Pages\timeline.jsx'
with open(jsx_path, 'r', encoding='utf-8') as f:
    code = f.read()

code = re.sub(r'mercury:\s*"data[^"]+",?', 'mercury: "/planets/mercury.png",', code)
code = re.sub(r'venus:\s*"data[^"]+",?', 'venus: "/planets/venus.png",', code)
code = re.sub(r'saturn:\s*"data[^"]+",?', 'saturn: "/planets/saturn.png",', code)

with open(jsx_path, 'w', encoding='utf-8') as f:
    f.write(code)

print("Updated timeline.jsx")
